packages/grid_client/scripts/mass_deployments.ts

```diff
@@ -129,78 +277,78 @@ async function main() {
+      vms.description = `Test deploying vm with name ${vm.name} - Batch ${batch + 1}`;
 
       batchVMs.push(vms);
     }
 
     const allTwinDeployments: TwinDeployment[] = [];
 
-    const deploymentPromises = batchVMs.map(async (vms, index) => {
+    const deploymentPromises = batchVMs.map(async vms => {
       try {
-        const [twinDeployments, _, __] = await grid3.machines._createDeployment(vms);
-        return { twinDeployments, batchIndex: index };
+        const [twinDeployments] = await grid3.machines._createDeployment(vms);
+        return twinDeployments;
       } catch (error) {
         log(`Error creating deployment for batch ${batch + 1}: ${error}`);
-        return { twinDeployments: null, batchIndex: index };
+        return [];
       }
     });
     console.time("Preparing Batch " + (batch + 1));
     const deploymentResults = await Promise.allSettled(deploymentPromises).then(results =>
       results.flatMap(r => (r.status === "fulfilled" ? r.value : [])),
     );
-    console.timeEnd("Preparing Batch " + (batch + 1));
-
-    for (const { twinDeployments } of deploymentResults) {
-      if (twinDeployments) {
-        allTwinDeployments.push(...twinDeployments);
-      }
-    }
-
-    try {
-      await grid3.machines.twinDeploymentHandler.handle(allTwinDeployments);
-      log(`Successfully handled and saved contracts for some twin deployments`);
-    } catch (error) {
-      errors.push(error);
-      failedCount += batchSize;
-      log(`Error handling contracts for twin deployments: ${error}`);
+    allTwinDeployments.push(...deploymentResults);
+    let batchSuccessfulNodes: Set<number> = new Set();
+    let batchFailedNodes: Set<number> = new Set();
+    if (allTwinDeployments.length > 0) {
+      const handlePromises = [handle(grid3, allTwinDeployments)];
+
+      const results = await Promise.allSettled(handlePromises);
+
+      results.forEach(result => {
+        if (result.status === "fulfilled") {
+          const { resultContracts, successfulNodes, failedNodes } = result.value;
+          batchSuccessfulNodes = successfulNodes;
+          batchFailedNodes = failedNodes;
+
+          successfulNodes.forEach(node => allSuccessfulNodes.push(node));
+          failedNodes.forEach(node => allFailedNodes.add(node));
+
+          log(`Successfully handled deployments for Batch ${batch + 1}`);
+        } else {
+          errors.push(`Error handling deployments for Batch ${batch + 1}: ${result.reason}`);
+        }
+      });
+
+      log(`Batch ${batch + 1} Summary:`);
+      log(`- Successful Deployments on Nodes: ${Array.from(batchSuccessfulNodes).join(", ")}`);
+      log(`- Failed Deployments on Nodes: ${Array.from(batchFailedNodes).join(", ")}`);
+      log(`- Successful Deployments: ${batchSuccessfulNodes.size}`);
+      log(`- Failed Deployments: ${batchFailedNodes.size}`);
+      log("---------------------------------------------");
```

Comment:
![image](https://github.com/user-attachments/assets/7e06fc90-1db6-446b-a316-4f75c114a0d1)
great, batch 1 summary is no longer empty. But can we please put a zero/ or a dash instead of empty size/nodes values? 
```suggestion
     log(`Batch ${batch + 1} Summary:`);
      log(
        `- Successful Deployments on Nodes: ${
          Array.from(batchSuccessfulNodes).length ? Array.from(batchSuccessfulNodes).join(", ") : "-"
        }`,
      );
      log(
        `- Failed Deployments on Nodes: ${
          Array.from(batchFailedNodes).length ? Array.from(batchFailedNodes).join(", ") : "-"
        }`,
      );
      log(`- Successful Deployments: ${batchSuccessfulNodes.size ?? 0}`);
      log(`- Failed Deployments: ${batchFailedNodes.size ?? 0}`);
      log("---------------------------------------------");
```